test/rubygems/test_gem_request.rb

```diff
@@ -491,22 +477,20 @@ def test_verify_certificate_message_UNABLE_TO_VERIFY_LEAF_SIGNATURE
   end
 
   def util_restore_version
-    Object.send :remove_const, :RUBY_ENGINE
-    Object.send :const_set,    :RUBY_ENGINE, @orig_RUBY_ENGINE if
-      defined?(@orig_RUBY_ENGINE)
+    Object.send :remove_const, :RUBY_ENGINE if defined?(RUBY_ENGINE)
```

Comment:
Why do we need this check (and similar below) now?

Reply:
Thanks. This is debug code before removing `test_user_agent_revision_missing` method. There is no reason we should check them. I removed all of guard conditions.